src/strategies/DSL/actions/marketSwapAction.ts

```diff
@@ -1,36 +1,50 @@
-import { MarketSwapActionArgs, PrimitiveArgs } from '@brinkninja/types'
+import { bigintToFeeAmount, toBigint, toTokenArgs } from '@brink-sdk/internal'
+import { UniV3Twap } from '@brink-sdk/oracles'
+import Token from '@brink-sdk/strategies/Token'
+import { MarketSwapActionArgs, PrimitiveArgs, TokenJSON } from '@brinkninja/types'
+
+const DEFAULT_TIME_INTERVAL = BigInt(1000)
 
 function marketSwapAction ({
   owner,
   tokenIn,
   tokenOut,
   tokenInAmount,
-  feePercent,
-  feeMinTokenOut
+  fee,
+  twapInterval = DEFAULT_TIME_INTERVAL,
+  twapFeePool
 }: MarketSwapActionArgs): PrimitiveArgs[] {
-  if (!feeMinTokenOut) {
-    feeMinTokenOut = 0
-  }
+  const twapFeePoolBN = twapFeePool ? toBigint(twapFeePool) : undefined // use helper function after merge from limitSwapAction
```

Comment:
I think here is where I would handle the undefined case. no need to run it through all these conversion fns if it's undefined